kfs-core/src/main/java/edu/arizona/kfs/fp/document/validation/impl/DisbursementVoucherDocumentPreRules.java

```diff
@@ -0,0 +1,125 @@
+package edu.arizona.kfs.fp.document.validation.impl;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.fp.businessobject.DisbursementVoucherPayeeDetail;
+import org.kuali.kfs.fp.document.DisbursementVoucherConstants;
+import org.kuali.kfs.fp.document.DisbursementVoucherDocument;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.kuali.rice.kew.api.WorkflowDocument;
+import org.kuali.rice.krad.document.Document;
+import org.kuali.rice.krad.util.GlobalVariables;
+
+import edu.arizona.kfs.fp.businessobject.DisbursementVoucherSourceAccountingLine;
+import edu.arizona.kfs.fp.businessobject.DisbursementVoucherSourceAccountingLineExtension;
+import edu.arizona.kfs.fp.service.DisbursementVoucherInvoiceService;
+import edu.arizona.kfs.sys.KFSConstants;
+import edu.arizona.kfs.sys.KFSKeyConstants;
+import edu.arizona.kfs.sys.KFSPropertyConstants;
+
+public class DisbursementVoucherDocumentPreRules extends org.kuali.kfs.fp.document.validation.impl.DisbursementVoucherDocumentPreRules {
+
+    private static transient volatile DisbursementVoucherInvoiceService disbursementVoucherInvoiceService;
+    private static transient volatile ConfigurationService configurationService;
+
+    private static DisbursementVoucherInvoiceService getDisbursementVoucherInvoiceService() {
+        if (disbursementVoucherInvoiceService == null) {
+            disbursementVoucherInvoiceService = SpringContext.getBean(DisbursementVoucherInvoiceService.class);
+        }
+        return disbursementVoucherInvoiceService;
+    }
+
+    private static ConfigurationService getConfigurationService() {
+        if (configurationService == null) {
+            configurationService = SpringContext.getBean(ConfigurationService.class);
+        }
+        return configurationService;
+    }
+
+    @Override
+    public boolean doPrompts(Document document) {
+        boolean result = super.doPrompts(document);
+
+        result &= checkDisbursementVoucherInvoiceNumberRequired((DisbursementVoucherDocument) document);

```

Comment:
I don't know. Without this, the system will allow a blank invoice number through, even though the DD definition has the required=true.